engine/schema/src/com/cloud/user/AccountVO.java

```diff
@@ -83,8 +86,16 @@ public AccountVO(String accountName, long domainId, String networkDomain, short
         this.domainId = domainId;
         this.networkDomain = networkDomain;
         this.type = type;
-        state = State.enabled;
+        this.state = State.enabled;
         this.uuid = uuid;
+        this.roleId = RoleType.getRoleByAccountType(null, type);
+    }
+
+    public AccountVO(String accountName, long domainId, String networkDomain, short type, Long roleId, String uuid) {
+        this(accountName, domainId, networkDomain, type, uuid);
```

Comment:
all these params can be final right? (probably missed that one in all the code above)